packages/eslint-plugin/src/rules/camelcase.ts

```diff
@@ -86,6 +101,27 @@ export default util.createRule<Options, MessageIds>({
       return false;
     }
 
+    /**
+     * Checks if the the node is a valid TypeScript generic type parameter
+     * @param node the node to be validated
+     * @returns true if the node is a TypeScript generic type parameter
+     * @private
+     */
+    function isTSTypeParameter(node: TSESTree.Node): boolean {
+      if (node.type === AST_NODE_TYPES.TSTypeParameter) {
+        return true;
+      }
+
+      if (node.type === AST_NODE_TYPES.TSTypeReference) {
+        return (
+          node.parent !== undefined &&
+          node.parent.type === AST_NODE_TYPES.TSTypeParameter
+        );
+      }
```

Comment:
I don't think we need to enforce the _usage_ of a generic is camelcase, right?
That should be covered by the previous check.